### PR TITLE
Adds profanity

### PR DESCRIPTION
**Pull request by nickgarlis**
[Repository link](https://github.com/nickgarlis/probot-profanity)

### CODE REVIEW

**Comment by hiimbex** [comment]
Thanks for this PR and building this app @nickgarlis ! Very cool idea.

I'm curious to hear more about why you chose to use the stale app to start your approach. This means relying on the scheduler extension for scheduled runs every hour as opposed to listening to issue and pull request and comment events and replying immediately.

Ideally, I think we want apps on the website to work as expected and demonstrate good practices. In this case, I think folks would expect the bot to respond immediately to profanity in issues and prs.

**Comment by nickgarlis** [comment]
Hi @hiimbex, thanks for taking the time to review my app.

My initial thought was that profanity bot should be able to scan for old issues and prs that could have profanity in them without having to wait for a specific event to take place. I also liked the `daysUntilClose` option in stale and thought that I could change it so that the bot could be configured to close profane issues/prs that haven't been edited within a specific number of days. For example if a user makes an issue that contains a forbidden word in it and doesn't edit it within x days then the issue will be closed.

You are right, people would actually expect it to react immediately. So I am guessing that wrapping the `close` functionality into a function that gets triggered periodically while making the rest respond to certain events would be a good way to keep this feature. 
What do you think ? The only downside to this is that users would have to trigger the bot at least once for old issues/prs to be marked/censored.

**Comment by bkeepers** [comment]
> The only downside to this is that users would have to trigger the bot at least once for old issues/prs to be marked/censored.

Your app can listen for the `installation` event and run against all repositories that it is installed on.

**Comment by nickgarlis** [comment]
@hiimbex Profanity bot can now respond to issue/pr events immediately using `context.payload`. However, the initial marking still works periodically just in case the bot runs out of actions upon the installation.

File: _apps/profanity.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+authors: [ nickgarlis ]
+repository: nickgarlis/probot-profanity
+host: https://probot-profanity.now.sh
+---
+
+## Usage
+
+1. **[Configure the GitHub App](https://github.com/apps/profanity)**
+2. Create `.github/profanity.yml` based on the following template
+3. It will start scanning for offensive issues and/or pull requests within an hour.
```
Comment (hiimbex) [inline _apps/profanity.md]: Maybe add "and automatically censor/mark any new issues/prs"

**Comment by nickgarlis** [comment]
@hiimbex I just added a few things to the docs since the bot can now be configured to have extra/exempt forbidden words and also detect profanity in other languages. Thanks !

**Comment by bkeepers** [comment]
Thanks @nickgarlis!

**Pull request merged**